### PR TITLE
[Improve] Add custom job id arg in client

### PR DESCRIPTION
**Pull request by Hisoka-X**
### Purpose of this pull request
This PR add a new param in `seatunnel.sh` named `--set-job-id`. User can use this param to fix job id when submit job. Not generate by SeaTunnel. This usage can facilitate third-party integration of SeaTunnel.

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
add new test.

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Review by hailin0 — APPROVED** [review]

**Review by lightzhao — APPROVED** [review]

**Review by wuchunfu — APPROVED** [review]

**Comment by Hisoka-X** [comment]
Thanks all, merged.

File: seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java [hunk 1]
```diff
@@ -67,8 +67,13 @@ public ClientJobExecutionEnvironment(
         this.seaTunnelHazelcastClient = seaTunnelHazelcastClient;
         this.jobClient = new JobClient(seaTunnelHazelcastClient);
         this.seaTunnelConfig = seaTunnelConfig;
-        this.jobConfig.setJobContext(
-                new JobContext(isStartWithSavePoint ? jobId : jobClient.getNewJobId()));
+        Long finalJobId;
+        if (isStartWithSavePoint || jobId != null) {
+            finalJobId = jobId;
```
Comment (liunaijie) [inline seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java:72]: @Hisoka-X hi, i relealize a problem, if user pass same job id with the running jobs, will it has issue? 
now i story some status in IMAP by job id. if use same job id, the status maybe impact?

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java:72]: Yes, the key point is that the ID cannot be repeated, which currently needs to be ensured by the user

Comment (Hisoka-X) [reply seatunnel-engine/seatunnel-engine-client/src/main/java/org/apache/seatunnel/engine/client/job/ClientJobExecutionEnvironment.java:72]: Fix by https://github.com/apache/seatunnel/pull/7021

**Pull request merged**